File: solutions/python/grains/1/grains.py
def square(number):
    if number > 64 or number < 1:
        raise ValueError("Square must be between 1 and 64")
    grains = 1
    for i in range(number-1):
        grains *= 2
    return grains


def total():
    grains = 1
    total = 0
    for i in range(64):
        total += grains
        if i < 64:
            grains *= 2
    return total

File: solutions/r/grains/1/grains.R
square <- function(n) {
    if (n > 64 || n < 1) {
        stop();
    }
    grains <- 1;
    if(n != 1) {
        for (i in seq(1,n-1)) { 
            grains <- grains * 2;
        }
    }
    return (grains)
}

total <- function() {
    grains <- 1;
    vetor <- c();
    for (i in seq(1,64)) {
        vetor <- c(vetor,grains);
        if (i < 64) {
            grains <- grains * 2;
        }
    }
    return (sum(vetor))
}

    